.github/workflows/test-pull-request.yml

```diff
@@ -0,0 +1,56 @@
+# Run Extension tests
+
+name: test-pull-request
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on pull requests that contain the label 'safe to test'
+  pull_request_target:
+    types: [ labeled ]
+
+permissions: {}    
+
+jobs:
+  test-smoke:
+    name: smoke-${{ matrix.version }}-${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test')
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [R2021b, R2023a]
+        os: [windows-latest, ubuntu-latest]
+    env:
+        MATHWORKS_SKIP_ACTIVATION: true
+        MLM_WEB_LICENSE: true
+        MLM_WEB_ID: ${{secrets.MLM_WEB_ID}}
+        MLM_WEB_USER_CRED: ${{secrets.MLM_WEB_USER_CRED}}
```

Comment:
Try to limit the scope of these env vars as much as possible. `MATHWORKS_SKIP_ACTIVATION` should only be necessary for `setup-matlab`:

```
      - name: Set up MATLAB
         uses: matlab-actions/setup-matlab@v1.2.3
         with:
           release: ${{ matrix.version }}
         env:
           MATHWORKS_SKIP_ACTIVATION: true
```

The MLM env vars should only be necessary for actions that run MATLAB (just `npm run tests`?).